mobile/src/people/people-room.tsx

```diff
@@ -25,13 +25,39 @@ const mapDispatchToProps = (dispatch: Dispatch<any>): EmployeesListDispatchProps
     onItemClicked: (employee: Employee) => dispatch( openEmployeeDetailsAction(employee))
 });
 
-export class PeopleRoomImpl extends React.Component<PeopleRoomProps & EmployeesListDispatchProps> {  
+export class PeopleRoomImpl extends React.Component<PeopleRoomProps & EmployeesListDispatchProps> {
+    public shouldComponentUpdate(nextProps: PeopleRoomProps & EmployeesListDispatchProps) {
+        if (this.props.onItemClicked !== nextProps.onItemClicked
+            || this.props.userEmployee !== nextProps.userEmployee
+        ) {
+            return true;
+        }
+
+        const predicate = (employee: Employee) => {
+            return this.props.userEmployee && employee.roomNumber === this.props.userEmployee.roomNumber;
+        };
+
+        const nextEmployees = nextProps.employees.employeesById.toArray().filter(predicate);
```

Comment:
Map supports filter, I don't think you need to call toArray() here (https://facebook.github.io/immutable-js/docs/#/Map/filter)

Reply:
Correct, but next line we are iterating over values and quit the loop once we found some new employee ref - what is a right way to rewrite it for Map collection?

Reply:
But I've just gave you the link on Map.filter function!

Reply:
And check out other comments first, the logic here might be corrupt.

Reply:
ok :)